app/src/main/java/com/gh4a/adapter/timeline/EventViewHolder.java

```diff
@@ -258,69 +259,94 @@ private CharSequence formatEvent(final IssueEvent event, final User user, int ty
         if (textBase == null) {
             textBase = mContext.getString(textResId, getUserLoginWithBotSuffix(user));
         }
+
         SpannableStringBuilder text = StringUtils.applyBoldTags(textBase, typefaceValue);
+        replaceCommitPlaceholder(text, event.commitId(), event.commitUrl());
+        replaceLabelPlaceholder(text, event.label());
+        replaceBotPlaceholder(text);
+        replaceSourcePlaceholder(text, event.source());
+        replaceTimePlaceholder(text, event.createdAt());
+        return text;
+    }
 
+    private void replaceCommitPlaceholder(SpannableStringBuilder text, String commitId, String commitUrl) {
         int pos = text.toString().indexOf("[commit]");
-        if (event.commitId() != null && pos >= 0) {
-            text.replace(pos, pos + 8, event.commitId().substring(0, 7));
-            text.setSpan(new IntentSpan(mContext, context -> {
-                // The commit might be in a different repo. The API doesn't provide
-                // that information directly, so get it indirectly by parsing the URL
-                String repoOwner = mRepoOwner;
-                String repoName = mRepoName;
-                String url = event.commitUrl();
-                if (url != null) {
-                    Matcher matcher = COMMIT_URL_REPO_NAME_AND_OWNER_PATTERN.matcher(url);
-                    if (matcher.find()) {
-                        repoOwner = matcher.group(1);
-                        repoName = matcher.group(2);
-                    }
-                }
-
-                return CommitActivity.makeIntent(context, repoOwner, repoName, event.commitId());
-            }), pos, pos + 7, 0);
-            text.setSpan(new TypefaceSpan("monospace"), pos, pos + 7, 0);
+        if (commitId == null || pos < 0) {
+            return;
+        }
+        // The commit might be in a different repo. The API doesn't provide
+        // that information directly, so get it indirectly by parsing the URL
+        String commitRepoOwner = mRepoOwner;
+        String commitRepoName = mRepoName;
+        boolean isCommitInDifferentRepo = false;
+        if (commitUrl != null) {
+            Matcher matcher = COMMIT_URL_REPO_NAME_AND_OWNER_PATTERN.matcher(commitUrl);
+            if (matcher.find()) {
+                commitRepoOwner = matcher.group(1);
+                commitRepoName = matcher.group(2);
+                isCommitInDifferentRepo = !mRepoOwner.equals(commitRepoOwner) || !mRepoName.equals(commitRepoName);
```

Comment:
Can do this outside of the if below (or even get rid of the variable completely).

Reply:
Done, I've kept the variable since it makes the condition more readable IMO.